### PR TITLE
Support virtual guests with public IPv6 addresses

### PR DESCRIPTION
**Pull request by minsikl**
Add IPv6 features to a virtual guest resource and fix minor bugs as follows: 
- Add ipv6_enabled, ipv6_address, ip_address_id6, and public_subnet6 to a virtual_guest schema.
- Remove createObject call and use PlaceOrder for virtual_guest creation.
- Fix user_metadata setting bug.

This PR provides function for #15.

### CODE REVIEW

File: docs/resources/softlayer_virtual_guest.md [hunk 1]
```diff
@@ -124,13 +124,22 @@ The following arguments are supported:
...
 
 * `id` - id of the virtual guest.
 * `ipv4_address` - Public IPv4 address of the virtual guest.
-* `ip_address_id_private` - Unique ID for the private ID address assigned to the virtual_guest.
+* `ip_address_id_private` - Unique ID for the private IPv4 address assigned to the virtual_guest.
 * `ipv4_address_private` - Private IPv4 address of the virtual guest.
-* `ip_address_id` - Unique ID for the public ID address assigned to the virtual_guest.
+* `ip_address_id` - Unique ID for the public IPv4 address assigned to the virtual_guest.
+* `ipv6_address` - Public IPv6 address of the virtual guest. It is provided when `ipv6_enabled` is `true`.
+* `ip_address_id6` - Unique ID for the public IPv6 address assigned to the virtual_guest. It is provided when `ipv6_enabled` is `true`.
```
Comment (renier) [inline docs/resources/softlayer_virtual_guest.md]: I would call this `ipv6_address_id`

File: docs/resources/softlayer_virtual_guest.md [hunk 2]
```diff
@@ -124,13 +124,22 @@ The following arguments are supported:
...
+* `public_subnet6` - Public IPv6 subnet. It is provided when `ipv6_enabled` is `true`.
```
Comment (renier) [inline docs/resources/softlayer_virtual_guest.md]: I would call this `public_ipv6_subnet`

**Review by renier — COMMENTED** [review]
Should we also wait for the ipv6 address to be set when we wait for the VM to come up?

File: docs/resources/softlayer_virtual_guest.md (hunk 1 above)
Comment (minsikl) [reply docs/resources/softlayer_virtual_guest.md]: done.

File: docs/resources/softlayer_virtual_guest.md (hunk 2 above)
Comment (minsikl) [reply docs/resources/softlayer_virtual_guest.md]: done.

**Comment by minsikl** [comment]
It looks like that we don't have to wait for the ipv6 address when we create a new VM.

**Pull request merged**